src/app/modules/main/wallet_section/activity/controller.nim

```diff
@@ -208,9 +208,12 @@ QtObject:
     if res.errorCode != ErrorCodeSuccess:
       error "error fetching activity entries: ", res.errorCode
       return
-
-    let entries = self.backendToPresentation(res.activities)
-    self.model.setEntries(entries, res.offset, res.hasMore)
+    
+    try: 
+      let entries = self.backendToPresentation(res.activities)
+      self.model.setEntries(entries, res.offset, res.hasMore)
+    except Exception as e:
```

Comment:
Better remove the exceptions in the function and transform them into errors. Ultimately, they did their job of signalling the issue but it took a lot of work to track. So I will only use Nim exceptions once we properly support reporting the stack trace. Errors logs are a better solution, even if they get lost in our noisy output.